clients/drcachesim/drpt2trace/pt2ir.cpp

```diff
@@ -62,6 +63,65 @@ namespace drmemtrace {
         }                                  \
     } while (0)
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* The subsequent C code is a patch for libipt. This patch introduces embedded APIs,
+ * enabling pt2ir_t to manage libipt's decoder's internal data structures and to
+ * facilitate stream decoding.
+ */
+
+/* An Intel PT packet decoder.
+ * The type mirrors the struct pt_packet_decoder found in
```

Comment:
@derekbruening what's the recommended approach for such structs. Do we add them to a separate file with some different copyright banner?

Reply:
Yes, it needs to remain separate and satisfy its license which requires keeping the copyright and license header.

Reply:
Noted.